### PR TITLE
samples: nrf_desktop: Stop advertising when turning off

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-463

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: samples/nrf_desktop/src/modules/ble_adv.c [hunk 1]
```diff
@@ -228,6 +231,7 @@ static void sp_grace_period_fn(struct k_work *work)
 	if (err) {
 		module_set_state(MODULE_STATE_ERROR);
 	} else {
+		state = STATE_OFF;
```
Comment (Qbicz) [inline samples/nrf_desktop/src/modules/ble_adv.c]: Here we shouldn't go to STATE_OFF but to IDLE. Remember that grace period is scheduled from `ble_adv_stop()` which is called e.g. when we get PEER_STATE_CONNECTED.

Comment (Qbicz) [reply samples/nrf_desktop/src/modules/ble_adv.c]: My understanding is that when we stop advertising, we go to IDLE (so advertising should be IDLE when BLE is connected) and that we go to OFF state only when power down event comes.

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/ble_adv.c]: discussed offline - grace_period timeout is scheduled from `remove_vendor_section` which is called on power down. When we reach line 234 it means that nobody connected to us during grace period and so we should continue with a system off.

**Comment by pdunaj** [comment]
hi @kapi-no , could you have a look?

**Comment by pdunaj** [comment]
Hi @anangl , could you give green tick or merge?

**Pull request merged**